src/sidebar/components/ThreadList.tsx

```diff
@@ -212,32 +213,34 @@ export default function ThreadList({ threads }: ThreadListProps) {
     }
   }, [store, newAnnotationTag]);
 
-  const mostRecentHighlightedAnnotationId = useMemo(() => {
-    const highlightedAnnos = allAnnotations.filter(
-      anno => anno.id && highlightedAnnotations.includes(anno.id),
-    );
-    // Get the annotation with most recent updated field, which contains a
-    // date in ISO format. This means their alphabetical and chronological
-    // orders match.
-    const mostRecentHighlightedAnnotation =
-      highlightedAnnos.reduce<Annotation | null>(
-        (mostRecent, current) =>
-          !mostRecent || mostRecent.updated < current.updated
-            ? current
-            : mostRecent,
-        null,
-      );
-
-    return mostRecentHighlightedAnnotation?.id;
-  }, [allAnnotations, highlightedAnnotations]);
-
-  // Scroll to the most recently highlighted annotation, unless creating/editing
+  const mostRelevantHighlightedAnnotationId = useMemo(() => {
+    // To determine which is the most relevant annotation, we first sort them
```

Comment:
This iterates over all annotations even if `highlightedAnnotations` is empty. I suggest adding an early exit in that case.